src/main/java/gyro/aws/acm/AcmCertificateFinder.java

```diff
@@ -119,12 +133,21 @@ protected List<CertificateDetail> findAws(AcmClient client, Map<String, String>
             filterBuilder = filterBuilder.keyUsageWithStrings(filters.get("key-algorithm"));
         }
 
+        List<String> certArns = new ArrayList<>();
         if (filterPresent) {
             builder = builder.includes(filterBuilder.build());
+
+            certArns.addAll(client.listCertificatesPaginator(builder.build()).certificateSummaryList()
+                .stream()
+                .map(CertificateSummary::certificateArn)
+                .collect(Collectors.toList()));
         }
 
-        return client.listCertificatesPaginator(builder.build()).certificateSummaryList().stream()
-            .map(o -> client.describeCertificate(r -> r.certificateArn(o.certificateArn())).certificate())
+        if (filters.containsKey("arn")) {
+            certArns.add(filters.get("arn"));
```

Comment:
Currently, this does an `OR` operation instead of an `AND` if there are 2 filters and one of them is the `arn`.
```suggestion
            certArns.removeIf(a -> !a.equals(filters.get("arn")));
```

Reply:
It is doing an `and` operation as the conditional block is not in an `else` state.
Adding the suggestion form Matt for using sets also takes care of the duplicates.